leaf-ui/scripts/main.js

```diff
@@ -262,6 +261,127 @@ $('#model-init-btn').on('click', function(){
 	switchToModellingMode(true);
 });
 
+//Cycle button onclick
+$('#cycledetect-btn').on('click', function(e){
+	//alert("cycle button clicked");
+	var analysis = new InputAnalysis();
+	var links = new InputLink();
+	var js_object = {};
+	var js_links = {};
+	js_object.analysis = getAnalysisValues(analysis);
+	jslinks = getLinks();
+	if(jslinks.length == 0){
+		swal("No cycle in the graph", "", "success");
+	}
+	else{
+		var verticies = js_object.analysis.elementList;
+		var links = jslinks;
+		//If there is no cycle, leave the color the way it was
+		if (cycleCheck(links, verticies) == false){
+			swal("No cycle in the graph", "", "success");
+			var elements = graph.getElements();
+			for (var i = 0; i < elements.length; i++){
+				var cellView  = elements[i].findView(paper);
+				if(cellView.model.attributes.type == "basic.Task"){
+					cellView.model.attr({'.outer': {'fill': '#92E3B1'}});
+				}
+				if(cellView.model.attributes.type == "basic.Goal"){
+					cellView.model.attr({'.outer': {'fill': '#FFCC66'}});
+				}
+				if(cellView.model.attributes.type == "basic.Resource"){
+					cellView.model.attr({'.outer': {'fill': '#92C2FE'}});
+				}
+				if(cellView.model.attributes.type == "basic.Softgoal"){
+					cellView.model.attr({'.outer': {'fill': '#FF984F'}});
+				}
+			}
+		}
+		else{
+			swal("Cycle in the graph", "", "error");
+			var elements = graph.getElements();
+			for (var i = 0; i < elements.length; i++){
+				var cellView  = elements[i].findView(paper);
+				if (recursiveStack[cellView.model.attributes.elementid] == true){
+					cellView.model.attr({'.outer': {'fill': 'red'}});
+				}
+				else{
+					if(cellView.model.attributes.type == "basic.Task"){
+						cellView.model.attr({'.outer': {'fill': '#92E3B1'}});
+					}
+					if(cellView.model.attributes.type == "basic.Goal"){
+						cellView.model.attr({'.outer': {'fill': '#FFCC66'}});
+					}
+					if(cellView.model.attributes.type == "basic.Resource"){
+						cellView.model.attr({'.outer': {'fill': '#92C2FE'}});
+					}
+					if(cellView.model.attributes.type == "basic.Softgoal"){
+						cellView.model.attr({'.outer': {'fill': '#FF984F'}});
+					}
+				}
+			}
+		}
+	}
+	var elements = graph.getElements();
+
+	js_object = null;
+	jslinks = null;
+})
+
+//Cycle-deteciton algorithm
+function cycleCheck(links, verticies){
```

Comment:
Need to include a comment: 

> Based on the Detect Cycle in a Directed Graph Algorithm discussed at 
http://www.geeksforgeeks.org/detect-cycle-in-a-graph/